synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -348,116 +349,127 @@ private String generatePart(final @NonNull String key, final @NonNull String val
-                        && response.getBody().has("message")) {
-                    throw new DataTransmissionException(response.getResponseCode(),
-                            response.getBody().getString("exception"),
-                            response.getBody().getString("error") + ": " + response.getBody().getString("message"));
-                } else {
-                    throw new DataTransmissionException(response.getResponseCode(), "unknown response attributes",
-                            response.getBody().toString());
-                }
-            } catch (final JSONException e) {
-                throw new ResponseParsingException(
-                        String.format("readResponse() failed: '%s'. Unable to read the http response.", e.getMessage()),
-                        e);
-            }
+        // Read response from connection
+        final int responseCode;
+        try {
+            responseCode = connection.getResponseCode();
+        } catch (final IOException e) {
+            throw new SynchronisationException(e);
+        }
+        final String responseBody = readResponseBody(connection);
+        final HttpResponse response = new HttpResponse(responseCode, responseBody);
+
+        // Handle known success responses
+        switch (responseCode) {
+            // Login Requests
+            case HttpURLConnection.HTTP_OK:
+                Log.d(TAG, "200: Login successful");
+                return response;
+            case HttpURLConnection.HTTP_CREATED:
+                Log.d(TAG, "201: Upload successful");
+                return response;
+        }
+
+        // Handle known error responses
+        switch (responseCode) {
+            case HttpURLConnection.HTTP_BAD_REQUEST:
+                Log.w(TAG, "400: Unknown error");
+                throw new BadRequestException(response.getBody());
+            case HttpURLConnection.HTTP_UNAUTHORIZED:
+                Log.w(TAG, "401: Bad credentials or missing authorization information");
+                throw new UnauthorizedException(response.getBody());
+            case HttpURLConnection.HTTP_FORBIDDEN:
+                Log.w(TAG, "403: The authorized user has no permissions to post measurements");
+                throw new ForbiddenException(response.getBody());
+            case HttpURLConnection.HTTP_CONFLICT:
+                Log.w(TAG, "409: The measurement already exists on the server.");
+                throw new ConflictException(response.getBody());
+            case HTTP_ENTITY_NOT_PROCESSABLE:
+                Log.w(TAG, "422: Multipart request is erroneous.");
+                throw new EntityNotParsableException(response.getBody());
+            case HttpURLConnection.HTTP_INTERNAL_ERROR:
+                Log.w(TAG, "500: Server reported internal error.");
+                throw new InternalServerErrorException(response.getBody());
         }
+
+        // Known response
```

Comment:
Alle in der API Spezifizierten Fehler werden hier "erkannt" und entsprechend getypt geworfen.